Профиль пользователя

NikitaGoroshko7 opened the pull request:
https://trello.com/c/ySpS3hkh/38-%D0%BF%D1%80%D0%BE%D1%84%D0%B8%D0%BB%D1%8C-%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8F

Has been added ProfileController
Has been created migrations and added extra properties for user
Has been added Views: Index, Settings, CreatedProjects and Balance
Has been added ViewModel for user setting's
Has been added a modal window for exit from profile and for delete profile
Has been added style and script for the modal window
Has been added folder Avatars for store user avatar's
Has been changed import.cs, was added using for work with files

NikitaGoroshko7 committed:
Профиль пользователя
https://trello.com/c/ySpS3hkh/38-%D0%BF%D1%80%D0%BE%D1%84%D0%B8%D0%BB%D1%8C-%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D0%BE%D0%B2%D0%B0%D1%82%D0%B5%D0%BB%D1%8F
Has been added ProfileController
Has been created migrations and added extra properties for user
Has been added Views: Index, Settings, CreatedProjects and Balance
Has been added ViewModel for user setting's
Has been added a modal window for exit from profile and for delete profile
Has been added style and script for the modal window
Has been added folder Avatars for store user avatar's
Has been changed import.cs, was added using for work with files

GalayM commented on CrowdFundingApplication/CrowdFundingApplication.csproj:
```diff
@@ -15,4 +15,8 @@
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="wwwroot\Avatars\" />
```
This folder is not accessible via a link without an authorization?

NikitaGoroshko7 replied:
Yes, not accessible
This folder just store user avatars

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/wwwroot/js/ModalWindow.js:
```diff
@@ -0,0 +1,33 @@
+    }
+});
+
+const openModal = (modal) => {
```
https://getbootstrap.com/docs/4.0/components/modal/

NikitaGoroshko7 replied:
In my case component bootstrap modal window did not work that's why I was using my custom modal window

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/wwwroot/js/ModalWindow.js:
```diff
@@ -0,0 +1,33 @@
+﻿let modal;
+document.addEventListener("click", (e) => {
```
Using jquery

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Views/Profile/Settings.cshtml:
```diff
@@ -0,0 +1,230 @@
+                    </div>
+                </div>
+
+                <script>
```
move to js file or js-section https://docs.microsoft.com/en-us/aspnet/core/mvc/views/layout?view=aspnetcore-6.0

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Views/Profile/Settings.cshtml:
```diff
@@ -0,0 +1,230 @@
+                                        <p><label for="file" name="Image" style="cursor: pointer;">Изменить</label></p>
+                                    </div>
+                                    <div class="col">
+                                        <p><label for="delete" style="cursor: pointer;" onclick="deleteFile()">Удалить</label></p>
```
Better link event via id + jquery

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Views/Profile/Balance.cshtml:
```diff
@@ -0,0 +1,53 @@
+
+    <div class="row mt-1">
+        <div class="col">
+            <div class="card shadow-2-strong card-registration" style="border-radius: 5px;">
```
click should not be in html

NikitaGoroshko7 replied:
I did not understand which exactly click?

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Models/User.cs:
```diff
@@ -2,6 +2,12 @@
+        public string Description { get; set; }
+        public int Balance { get; set; }
+        public int Age { get; set; }
+        public string Sex { get; set; }
```
Using enum

NikitaGoroshko7 replied:
But I need all this properties for user

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Models/User.cs:
```diff
@@ -2,6 +2,12 @@
-
+        public string Avatar { get; set; }
+        public string Description { get; set; }
+        public int Balance { get; set; }
```
If it many - better decimal

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Models/User.cs:
```diff
@@ -2,6 +2,12 @@
     public class User : IdentityUser
     {
-
+        public string Avatar { get; set; }
```
Avatar -> AvatarFileName

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Models/EditProfileViewModel.cs:
```diff
@@ -0,0 +1,31 @@
+
+    [StringLength(100, MinimumLength = 10, ErrorMessage = "Длина строки должна быть от 10 до 100 символов !")]
+    public string Description { get; set; }
+
```
???

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Models/EditProfileViewModel.cs:
```diff
@@ -0,0 +1,31 @@
+
+public class EditProfileViewModel
+{
+    [EmailAddress (ErrorMessage = "Неверный email адрес !")]
```
Неверный email адрес ! -> Неверный email адрес! 
In all cases

NikitaGoroshko7 replied:
If I will not use this attribute then all my issues will be in English, that's why I used it

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+    public IActionResult Settings() => View();
+
+    [HttpPost]
+    public async Task<IActionResult> Settings(EditProfileViewModel model, IFormFile Image)
```
It is work? Did you test?

NikitaGoroshko7 replied:
Yes it works

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Views/Profile/Index.cshtml:
```diff
@@ -0,0 +1,146 @@
+﻿@model CrowdFundingApplication.Models.User
```
You cannot use domain models for mapping. In all cases.

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+    {
+        var user = await _userManager.FindByNameAsync(HttpContext.User.Identity.Name);
+        await _userManager.DeleteAsync(user);
+        return RedirectToAction("MainPage", "Home");
```
Logout?

NikitaGoroshko7 replied:
No, this is action which is delete user from database

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+        if (image is not null)
+        {
+            string uploadsFolder = Path.Combine(_webHostEnvironment.WebRootPath, "Avatars");
+            uniqueFileName = Guid.NewGuid().ToString() + "_" + image.FileName;
```
We discussed it!

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+
+        if (image is not null)
+        {
+            string uploadsFolder = Path.Combine(_webHostEnvironment.WebRootPath, "Avatars");
```
string -> var

There must be 1 style or we always write styles or always var

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+            }
+
+            //update dates for user
+            if (model.Age is not null) user.Age = (int)model.Age;
```
if Age = null throw exeprion

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+
+            //update dates for user
+            if (model.Age is not null) user.Age = (int)model.Age;
+            if (model.Email is not null) user.Email = model.Email;
```
You can't just change the email - look at this table in the database and you will understand what I mean.

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+
+            if (model.NewPassword is not null)//update password for user
+            {
+                var _passwordValidator = HttpContext.RequestServices.GetService(typeof(IPasswordValidator<User>)) as IPasswordValidator<User>;
```
Use DI via constructor

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+            {
+                var _passwordValidator = HttpContext.RequestServices.GetService(typeof(IPasswordValidator<User>)) as IPasswordValidator<User>;
+                var _passwordHasher = HttpContext.RequestServices.GetService(typeof(IPasswordHasher<User>)) as IPasswordHasher<User>;
+                IdentityResult result = await _passwordValidator.ValidateAsync(_userManager, user, model.NewPassword);
```
http://dotnet-tutorials.net/identity/update-user-password

NikitaGoroshko7 replied:
Has been edited

GalayM commented on CrowdFundingApplication/Controllers/ProfileController.cs:
```diff
@@ -0,0 +1,94 @@
+
+    public async Task<IActionResult> Index()
+    {
+        var user = await _userManager.FindByNameAsync(HttpContext.User.Identity.Name);
```
HttpContext.User.Identity.Name => User.Identity.Name

NikitaGoroshko7 replied:
Has been edited

NikitaGoroshko7 committed:
Has been edited ProfileController
Has been updated column names
Has been edited and added models
Has been edited Views
Has been deleted unnecessary files

GalayM commented on CrowdFundingApplication/Models/UserViewModel.cs:
```diff
@@ -0,0 +1,15 @@
+    public string AvatarFileName { get; set; }
+    public string Email { get; set; }
+    public string PhoneNumber { get; set; }
+    public int Age { get; set; }
```
int -> Sex

NikitaGoroshko7 replied:
Has been edited

NikitaGoroshko7 committed:
Has been deleted bug with modal window

NikitaGoroshko7 committed:
Has been edited enum Sex and edited View
